modin/experimental/core/execution/native/implementations/hdk_on_native/dataframe/dataframe.py

```diff
@@ -2010,18 +2118,26 @@ def _can_execute_arrow(self):
         """
         if isinstance(self._op, FrameNode):
             return self._has_arrow_table()
-        elif isinstance(self._op, MaskNode):
-            return (
-                self._op.row_labels is None and self._op.input[0]._can_execute_arrow()
-            )
-        elif isinstance(self._op, TransformNode):
-            return (
-                self._op.is_simple_select() and self._op.input[0]._can_execute_arrow()
-            )
-        elif isinstance(self._op, UnionNode):
-            return all(frame._can_execute_arrow() for frame in self._op.input)
-        else:
-            return False
+
+        stack = [self]
+        while stack:
+            frame = stack.pop()
+            if isinstance(frame._op, FrameNode):
+                if not frame._has_arrow_table():
+                    return False
+            elif isinstance(frame._op, MaskNode):
+                if frame._op.row_labels is not None:
+                    return False
```

Comment:
there might be something wrong here, because [this piece of code](https://github.com/AndreyPavlenko/modin/blob/2268a60b737b19a0b74a2712eac5f8e8f61a8a33/modin/experimental/core/execution/native/implementations/hdk_on_native/dataframe/dataframe.py#L2167-L2168) claims that arrow can process a MaskNode with row labels specified and then here we claim that it can't

Reply:
This condition was in the original code - https://github.com/AndreyPavlenko/modin/blob/master/modin/experimental/core/execution/native/implementations/hdk_on_native/dataframe/dataframe.py#L1959 .
Actually, row_labels is not yet supported - https://github.com/AndreyPavlenko/modin/blob/master/modin/experimental/core/execution/native/implementations/hdk_on_native/dataframe/dataframe.py#L383.